redbox-core/redbox/transform.py

```diff
@@ -85,33 +103,23 @@ def structure_documents_by_group_and_indices(docs: list[Document]) -> DocumentSt
 
     The document_uuid is taken from the Document metadata directly.
     """
-    result = DocumentState()
-    current_group = DocumentGroup()
-    current_group_indices: list[int] = []
-    current_filename: str | None = None
 
-    for d in docs:
-        is_not_same_filename = d.metadata["uri"] != current_filename
-        is_not_none_filename = current_filename is not None
-        is_not_consecutive = d.metadata["index"] - 1 != (
-            current_group_indices[-1] if current_group_indices else d.metadata["index"]
-        )
-        if (is_not_same_filename and is_not_none_filename) or (is_not_consecutive and is_not_none_filename):
-            # Generate a deterministic hash for the previous document and its indices
-            group_id = create_group_uuid(current_filename, current_group_indices)
-            result.groups[group_id] = current_group
-
-            current_group = DocumentGroup()
-            current_group_indices: list[int] = []
-
-        current_group[d.metadata["uuid"]] = d
-        current_group_indices.append(d.metadata["index"])
-        current_filename = d.metadata["uri"]
+    groups = []
+    current_group = [docs[0]]
```

Comment:
is there always at least one document?

Reply:
[d487abf](https://github.com/i-dot-ai/redbox/pull/1210/commits/d487abfb6a75780e9197400adab4b4ce8a5870d4)